/*
 * This file was automatically generated by EvoSuite
 * Sat May 09 18:22:46 GMT 2020
 */

package ElevatorSystem;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import ElevatorSystem.Elevator;
import ElevatorSystem.Environment;
import ElevatorSystem.Floor;
import ElevatorSystem.Person;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class Floor_ESTest extends Floor_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Environment environment0 = new Environment(1);
      int int0 = (-522);
      Floor floor0 = new Floor(environment0, (-522));
      floor0.callElevator();
      String string0 = "H-B.C";
      int int1 = 3163;
      Person person0 = new Person("H-B.C", 3163, 0, 100, environment0);
      person0.leaveElevator();
      person0.leaveElevator();
      floor0.addWaitingPerson(person0);
      floor0.hasCall();
      floor0.addWaitingPerson(person0);
      Elevator elevator0 = null;
      // Undeclared exception!
      try { 
        floor0.processWaitingPersons((Elevator) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("ElevatorSystem.Floor", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      int int0 = 3342;
      Environment environment0 = new Environment(3342);
      Floor floor0 = new Floor(environment0, 2696);
      Person person0 = new Person("", 3342, 0, 2696, environment0);
      floor0.addWaitingPerson(person0);
      floor0.hasCall();
      floor0.callElevator();
      String string0 = null;
      Floor[] floorArray0 = new Floor[5];
      floorArray0[0] = floor0;
      floorArray0[1] = floor0;
      floorArray0[2] = floor0;
      floorArray0[3] = floor0;
      floorArray0[4] = floor0;
      environment0.floors = floorArray0;
      Elevator elevator0 = new Elevator(environment0, true);
      // Undeclared exception!
      try { 
        floor0.processWaitingPersons(elevator0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 2696
         //
         verifyException("ElevatorSystem.Elevator", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Environment environment0 = null;
      Floor floor0 = new Floor((Environment) null, 0);
      floor0.callElevator();
      floor0.callElevator();
      floor0.hasCall();
      floor0.getFloorID();
      floor0.hasCall();
      floor0.reset();
      floor0.getFloorID();
      Person person0 = null;
      try {
        person0 = new Person("g@~TJ@3h|:", 0, 0, 0, (Environment) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("ElevatorSystem.Person", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      Environment environment0 = new Environment(1751);
      Floor floor0 = new Floor(environment0, (-1));
      floor0.callElevator();
      Floor[] floorArray0 = new Floor[4];
      floorArray0[0] = floor0;
      floorArray0[1] = floor0;
      floorArray0[2] = floor0;
      floorArray0[3] = floor0;
      environment0.floors = floorArray0;
      floor0.reset();
      floor0.getFloorID();
      floor0.callElevator();
      assertTrue(floor0.hasCall());
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      Environment environment0 = new Environment(0);
      Floor floor0 = new Floor(environment0, 0);
      assertFalse(floor0.hasCall());
      
      floor0.callElevator();
      assertTrue(floor0.hasCall());
      
      floor0.callElevator();
      floor0.getFloorID();
      Person person0 = null;
      try {
        person0 = new Person("ElevatorSystem.Floor", (-1978), 0, 0, environment0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 0
         //
         verifyException("ElevatorSystem.Environment", e);
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      Environment environment0 = new Environment(1823);
      Floor floor0 = new Floor(environment0, 1823);
      floor0.callElevator();
      floor0.reset();
      Elevator elevator0 = new Elevator(environment0, true, 66, false);
      floor0.processWaitingPersons(elevator0);
      elevator0.weight = 0;
      Person person0 = new Person("ElevatorSystem.Floor", 1823, 0, 66, environment0);
      person0.leaveElevator();
      floor0.addWaitingPerson(person0);
      person0.leaveElevator();
      floor0.addWaitingPerson(person0);
      person0.leaveElevator();
      floor0.callElevator();
      floor0.addWaitingPerson(person0);
      floor0.callElevator();
      floor0.processWaitingPersons(elevator0);
      floor0.reset();
      floor0.hasCall();
      floor0.getFloorID();
      floor0.getFloorID();
      floor0.callElevator();
      floor0.processWaitingPersons(elevator0);
      floor0.reset();
      Person person1 = new Person("ElevatorSystem.Floor", 0, 0, 0, environment0);
      floor0.addWaitingPerson(person1);
      floor0.addWaitingPerson(person1);
      int int0 = floor0.getFloorID();
      assertTrue(floor0.hasCall());
      assertEquals(1823, int0);
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      int int0 = 0;
      Environment environment0 = new Environment(0);
      Floor[] floorArray0 = new Floor[8];
      Environment environment1 = new Environment(1774);
      Floor floor0 = environment1.getFloor(0);
      floorArray0[0] = floor0;
      Environment environment2 = new Environment(0);
      Floor floor1 = new Floor(environment2, 1774);
      floorArray0[1] = floor1;
      Floor floor2 = new Floor(environment0, 0);
      floorArray0[2] = floor2;
      Floor floor3 = new Floor(environment2, 0);
      floorArray0[3] = floor3;
      Floor floor4 = new Floor(environment2, 178);
      floorArray0[4] = floor4;
      // Undeclared exception!
      try { 
        environment2.getFloor(0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 0
         //
         verifyException("ElevatorSystem.Environment", e);
      }
  }

  @Test(timeout = 4000)
  public void test7()  throws Throwable  {
      int int0 = (-172);
      Environment environment0 = null;
      try {
        environment0 = new Environment((-172));
        fail("Expecting exception: NegativeArraySizeException");
      
      } catch(NegativeArraySizeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("ElevatorSystem.Environment", e);
      }
  }
}
